lib/settings-storage-service.ts

```
@@ -0,0 +1,50 @@
+  reset(): Promise<Settings>
+}
+
+const createSettingsStorageService = (): SettingsStorageService => {
+  const service: SettingsStorageService = {
+    async set(settings: Settings): Promise<Settings> {
+      try {
+        await storage.setItem("settings", settings)
+      } catch (error) {
+        console.error("Failed to set settings:", error)
+      } finally {
+        return this.get()
```

Biome (1.9.4)

[error] Unsafe usage of 'return'.

'return' in 'finally' overwrites the control flow statements inside 'try' and 'catch'.

(lint/correctness/noUnsafeFinally)